### PR TITLE
Updated docs for the improved Google OIDC connector

### LINKED ISSUE #8942: Google Workspace SSO Missing Step
To use Google SSO with Teleport, you must enable the Admin SDK API in your GCP project. This step is not mentioned in [the docs](https://goteleport.com/docs/enterprise/sso/google-workspace/#google-workspace-as-sso-for-ssh). 

Presently, the only way a user would know to do this would be by looking at the error auth log that tells the user they must enable the Admin SDK API.

### PR DESCRIPTION
**Pull request by espadolini**
Updated docs for
* #9697

### CODE REVIEW

**Comment by espadolini** [comment]
Should I temporarily remove the video banner until a new video is made?

File: docs/pages/enterprise/sso/google-workspace.mdx [hunk 1]
```diff
@@ -27,64 +27,81 @@ to define policies like:
...
+  - i.e. a dedicated account `ben-ops@example.com` as opposed to your daily `ben@example.com` account.
+- Ability to create a Google Cloud Platform project.
+  - This might require signing up to GCP, but the created project will not require using any paid services. It’s just a side effect of Google Workspace and GCP being closely related.
+- Ability to setup Google Workspace groups.
+
+## Configuring Google Workspace
+
+The setup will consist of creating a new project on Google Cloud Platform,
+configuring its OAuth consent screen, creating an OAuth client ID to allow
+Google Workspace users to log in to your Teleport, and creating a service
```
Comment (zmb3) [inline docs/pages/enterprise/sso/google-workspace.mdx:41]:
```suggestion
Google Workspace users to log in to your Teleport cluster, and creating a service
```

**Review by zmb3 — APPROVED** [review]

File: docs/pages/enterprise/sso/google-workspace.mdx [hunk 2]
```diff
@@ -140,6 +147,33 @@ Create the connector using `tctl` tool:
...
+
+First, fetch the connector data:
+```bsh
+$ umask 077
+$ tctl get --with-secrets oidc/connectorname > connector.yaml
+```
+
+then edit `connector.yaml` to change the version number from `v2` to `v3`, and then update the connector:
+
+```bsh
```
Comment (ptgott) [inline docs/pages/enterprise/sso/google-workspace.mdx:160]: Let's use the `code` snippet here so users can copy individual commands

File: docs/pages/enterprise/sso/google-workspace.mdx [hunk 3]
```diff
@@ -140,6 +147,33 @@ Create the connector using `tctl` tool:
 $ tctl create gworkspace-connector.yaml
 ```
 
+If you have a configured connector from a version of Teleport older than 8.1.1, you can upgrade your connector from `v2` to `v3`:
+
+First, fetch the connector data:
+```bsh
```
Comment (ptgott) [inline docs/pages/enterprise/sso/google-workspace.mdx:153]: Let's use the `code` snippet here so users can copy individual commands

File: docs/pages/enterprise/sso/google-workspace.mdx [hunk 4]
```diff
@@ -140,6 +147,33 @@ Create the connector using `tctl` tool:
...
+Then, to start fetching transitive groups instead of just direct groups, edit
+the [domain-wide
+delegation](https://admin.google.com/ac/owl/domainwidedelegation) for your
+service account and swap out the OAuth scopes for
+`https://www.googleapis.com/auth/cloud-identity.groups.readonly`. To undo the
+change, remove that scope and add
+`https://www.googleapis.com/auth/admin.directory.group.readonly` again.
+
+While a `v3` connector is configured, you can no longer downgrade Teleport to a
+version before 8.1.1; before such a downgrade, follow the above instructions and
```
Comment (ptgott) [inline docs/pages/enterprise/sso/google-workspace.mdx:174]:
```suggestion
version before 8.1.1. Before such a downgrade, follow the above instructions and
```

File: docs/pages/enterprise/sso/google-workspace.mdx [hunk 5]
```diff
@@ -27,95 +27,107 @@ to define policies like:
...
-
-### Download Service Account JSON
-
-![Create OAuth Creds](../../../img/gsuite/gsuite-5f-download-json.png)
-
-This JSON file will need to be uploaded to the Authentication server, and will be later referenced by
-the OIDC Connector, under `google_service_account_uri` or inline with `google_service_account`.
+- An Enterprise version of Teleport downloaded from [https://dashboard.gravitational.com/](https://dashboard.gravitational.com/).
+- A Google Workspace Super administrator account. We recommend setting up a separate super admin account with 2FA as opposed to granting your daily user super admin privileges.
+   Use a dedicated account, e.g., `ben-ops@example.com`, as opposed to your daily `ben@example.com` account.
```
Comment (ptgott) [inline docs/pages/enterprise/sso/google-workspace.mdx:32]:
```suggestion

Use a dedicated account, e.g., `ben-ops@example.com`, as opposed to your daily `ben@example.com` account.
```
I think this bullet is a bit long, so I've suggested adding another line.

File: docs/pages/enterprise/sso/google-workspace.mdx [hunk 6]
```diff
@@ -27,95 +27,107 @@ to define policies like:
...
-
-Within GSuite to access the Manage API client access go to Security -> Settings.  Navigate to Advanced Settings and open Manage API client access.  Put the client ID in the Client Name field and the below permissions in the API scopes as a single comma separated line.  Press Authorize.
+- Configure [domain-wide
+  delegation](https://admin.google.com/ac/owl/domainwidedelegation) for your
+  newly-created service account and add either the
+  `https://www.googleapis.com/auth/cloud-identity.groups.readonly` scope or the
+  `https://www.googleapis.com/auth/admin.directory.group.readonly` scope. The
+  scope granted to the service account will determine if Teleport will fetch
+  both direct and indirect groups or just direct groups, respectively.
+  ![domain-wide delegation](../../../img/googleoidc/domainwidedelegation.png)
```
Comment (ptgott) [inline docs/pages/enterprise/sso/google-workspace.mdx:109]: I had a little trouble connecting the screenshot with the text. Maybe add one more sentence (in a new paragraph) explaining what this image shows?

**Review by ptgott — APPROVED** [review]
I had two remaining nits, but I've approved so my review doesn't block you.

File: docs/pages/enterprise/sso/google-workspace.mdx (hunk 5 above)
Comment (espadolini) [reply docs/pages/enterprise/sso/google-workspace.mdx:32]: Honestly, "we recommend setting up a separate super admin account" should be enough; using something like `ben-ops` is probably a bad idea compared to `teleport-auth-helper` anyway, so I just removed the example.

**Pull request merged**